### PR TITLE
Add configuration options to tune `SearchIndexingBufferedSender<T>`.

### LINKED ISSUE #16291: [Flaky test] AddsDelay
```
Error message
Expected: less than or equal to 79
But was:  80


Stack trace
   at Azure.Search.Documents.Tests.ManualRetryDelayTests.AddsDelay() in /home/vsts/work/1/s/sdk/search/Azure.Search.Documents/tests/Batching/ManualRetryDelayTests.cs:line 29

```

### PR DESCRIPTION
**Pull request by tg-msft**
Also adds better behavior around swallowing notification errors and fixes #16291 to make tests less flaky.

### CODE REVIEW

File: sdk/search/Azure.Search.Documents/src/Batching/Publisher{T}.cs [hunk 1]
```diff
@@ -134,9 +134,16 @@ public int IndexingActionsCount
         /// The number of bytes to use when tuning the behavior of the
         /// publisher.
         /// </param>
-        /// <param name="retryCount">
+        /// <param name="maxRetries">
         /// The number of times to retry a failed document.
         /// </param>
+        /// <param name="retryDelay">
+        /// The initial retry delay on which to base calculations for a
+        /// backoff-based approach.
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/src/Batching/Publisher{T}.cs:142]: I'd perhaps use more plain language here, like "The initial retry delay. Subsequent retries may take longer up to the <paramref name="maxRetryDelay"/>."

File: sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingBufferedSender{T}.cs [hunk 2]
```diff
@@ -331,10 +333,20 @@ private async Task GetKeyFieldAccessorAsync(bool async, CancellationToken cancel
...
+        {
+            try
+            {
+                await ActionAddedAsync
+                    .RaiseAsync(action, cancellationToken)
+                    .ConfigureAwait(false);
+            }
+            catch
+            {
+                // TODO: Log any exceptions raised from async events that we
```
Comment (heaths) [inline sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingBufferedSender{T}.cs:346]: Is there a tracking issue on this? I assume to add an EventSource?

**Review by heaths — APPROVED** [review]

File: sdk/search/Azure.Search.Documents/src/Batching/Publisher{T}.cs (hunk 1 above)
Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/src/Batching/Publisher{T}.cs:142]: Fixed

Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/src/Batching/Publisher{T}.cs:142]: Fixed

File: sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingBufferedSender{T}.cs (hunk 2 above)
Comment (tg-msft) [reply sdk/search/Azure.Search.Documents/src/Batching/SearchIndexingBufferedSender{T}.cs]: Fixed

**Pull request merged**